src/query/api/v1/handler/prom/prom.go

```diff
@@ -41,26 +41,35 @@ func init() {
 // Options defines options for PromQL handler.
 type Options struct {
 	PromQLEngine *promql.Engine
+	instant bool
+}
+
+// WithInstant returns new Options copy with given instant value
+func (o Options) WithInstant(instant bool) Options {
+	return Options{
+		PromQLEngine: o.PromQLEngine,
+		instant:      instant,
+	}
```

Comment:
I've seen it done like this in quite a few places:

https://github.com/m3db/m3/blob/166bdb36cdb589caa67c3ef2df0b13c02b1d80de/src/dbnode/ratelimit/options.go#L50-L54

Reply:
I think for configuration types immutability should be the default. Also, we are constructing two different read handlers and they need to have different options in order to work properly so this method just makes this much easier to do and more readable.

Reply:
I agree about the immutability. In the example above, `opts := *o` creates a copy of the struct, hence the original is kept intact. 

Also, a drawback of constructing a copy by listing all the values inside `Options{ /* ... */ }` is that it doesn't scale well with the growth of the structure — once there are more fields, it will be cumbersome to list them all. Especially if there are multiple methods that create a mutated copy of the original.

Reply:
Refactored to use functional options pattern.